packages/core/src/components/stepper/stepper.tsx

```diff
@@ -0,0 +1,113 @@
+import { Component, Host, Prop, Watch, h } from '@stencil/core'
+import DOMPurify from 'dompurify'
+
+import { isMaxTablet } from '../../utils/screens'
+
+@Component({
+  tag: 'atom-stepper',
+  styleUrl: 'stepper.scss',
+  shadow: true,
+})
+export class AtomStepper {
+  @Prop({ mutable: true }) steps:
+    | string
+    | {
+        title: string
+        completed: boolean
+      }[] = []
+  @Prop() activeStep: number
+  @Prop() stepTitle?: string
+  @Prop() disabledUnrenderedStep?: boolean = true
+
+  private parsedSteps: {
+    title: string
+    completed: boolean
+  }[] = []
+
+  @Watch('steps')
+  parseSteps(newVal) {
+    this.parsedSteps = typeof newVal === 'string' ? JSON.parse(newVal) : newVal
+  }
+
+  componentWillLoad() {
+    if (typeof this.steps === 'string') {
+      this.parsedSteps = JSON.parse(this.steps)
+    } else {
+      this.parsedSteps = this.steps
+    }
+  }
+
+  private readonly isCompleted = (completed: boolean) =>
+    completed ? 'completed' : ''
+
+  private readonly isLastActive = (index: number) => {
+    const currentStep = this.parsedSteps[index]
+    const nextStep = this.parsedSteps[index + 1]
+
+    const currentStepIsTheLastCompleted =
+      currentStep?.completed && nextStep && !nextStep.completed
+
+    return currentStepIsTheLastCompleted ? 'last' : ''
+  }
+
+  render() {
+    const actualStep = this.parsedSteps[this.activeStep]
+    const completedStep = actualStep?.completed
+
+    return (
+      <Host role='stepper'>
+        {isMaxTablet() ? (
+          <div>
+            {actualStep && (
+              <div
+                class={`atom-stepper-mobile ${completedStep ? 'completed' : ''}`}
+              >
+                <div class='number'>
+                  {completedStep ? (
+                    <atom-icon icon='check' color='white' size={26} />
+                  ) : (
+                    this.activeStep + 1
+                  )}
+                </div>
+                <div>
+                  <div innerHTML={DOMPurify.sanitize(this?.stepTitle)}></div>
+                  <span class='title'>{actualStep.title}</span>
+                </div>
+              </div>
+            )}
+          </div>
+        ) : (
+          <ul class='atom-stepper'>
+            {this.parsedSteps.map(({ title, completed }, index) => {
+              const isActive = index === this.activeStep
+              const shouldDisabled =
+                this.disabledUnrenderedStep && index > this.activeStep
```

Comment:
Question: Should the icon button be interactive when disabledUnrenderedStep is false?

Reply:
When true.

disabledUnrenderedStep is a prop that disables all steps after the current step (index > this.activeStep). But this is an optional feature, so the prop must be true, and the index must be greater than the current step.

![image](https://github.com/user-attachments/assets/cd991ce9-90a3-4f72-a21f-125823862845)

Reply:
Thanks for explaining, @gabrielduete. Guess I didn't make myself clear; I mean, should the icon button be **clickable** (click event to do something like go to next step) when disabledUnrenderedStep is `false`? If not, what is the diff between them?

Reply:
Hey man, I saw the Figma. I guess we don't need this prop, as it looks like we shouldn't have a disabledUnrenderedStep=false scenario. Did I get it right?

Reply:
> Thanks for explaining, @gabrielduete. Guess I didn't make myself clear; I mean, should the icon button be clickable (click event to do something like go to next step) when disabledUnrenderedStep is false? If not, what is the diff between them?

Yeah! The difference is only visual, it has those “disabled” styles that we use in components. The option to disable the button depends on the logic of the page that will use the component.

> Hey man, I saw the Figma. I guess we don't need this prop, as it looks like we shouldn't have a disabledUnrenderedStep=false scenario. Did I get it right?

Talking to the UX team, we saw that this scenario does not exist today, but it would be good to leave it in case we need it.

Reply:
> Thanks for explaining, @gabrielduete. Guess I didn't make myself clear; I mean, should the icon button be clickable (click event to do something like go to next step) when disabledUnrenderedStep is false? If not, what is the diff between them?

Yeah! The difference is only visual, it has those “disabled” styles that we use in components. The option to disable the button depends on the logic of the page that will use the component.

> Hey man, I saw the Figma. I guess we don't need this prop, as it looks like we shouldn't have a disabledUnrenderedStep=false scenario. Did I get it right?

Talking to the UX team, we saw that this scenario does not exist today, but it would be good to keep it in case we need it.

Reply:
ok man! Thanks